t/e2e/services/debug_api_test.go

```diff
@@ -0,0 +1,152 @@
+	"fmt"
+	"io/ioutil"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	whisper "github.com/ethereum/go-ethereum/whisper/whisperv6"
+	"github.com/status-im/status-go/node"
+	"github.com/status-im/status-go/params"
+	"github.com/stretchr/testify/suite"
+
+	. "github.com/status-im/status-go/t/utils"
+)
+
+func TestDebugAPISuite(t *testing.T) {
+	s := new(DebugAPISuite)
+	s.upstream = false
+	suite.Run(t, s)
+}
+
+func TestDebugAPISuiteUpstream(t *testing.T) {
+	s := new(DebugAPISuite)
+	s.upstream = true
+	suite.Run(t, s)
+}
+
+type DebugAPISuite struct {
+	BaseJSONRPCSuite
+	upstream bool
+}
+
+func (s *DebugAPISuite) TestAccessibleDebugAPIsUnexported() {
+	if s.upstream && GetNetworkID() == params.StatusChainNetworkID {
+		s.T().Skip()
+		return
+	}
+
+	err := s.SetupTest(s.upstream, false, false)
+	s.NoError(err)
+	// Debug APIs should be unavailable
+	s.AssertAPIMethodUnexported("debug_postSync")
+	err = s.Backend.StopNode()
+	s.NoError(err)
+
+	err = s.SetupTest(s.upstream, false, true)
+	s.NoError(err)
+	defer func() {
+		err := s.Backend.StopNode()
+		s.NoError(err)
+	}()
+	// Debug APIs should be available
+	s.AssertAPIMethodExported("debug_postSync")
+}
+
+func (s *DebugAPISuite) TestDebugPostSyncSuccess() {
+	// Test upstream if that's not StatusChain
+	if s.upstream && GetNetworkID() == params.StatusChainNetworkID {
+		s.T().Skip()
+		return
+	}
+
+	err := s.SetupTest(s.upstream, false, true)
+	s.NoError(err)
+	defer func() {
+		err := s.Backend.StopNode()
+		s.NoError(err)
+	}()
+
+	s.addPeerToCurrentNode()
+	symID := s.generateSymKey()
+	result := s.sendPostConfirmMessage(symID)
+
+	var r struct {
+		Error struct {
+			Message string `json:"message"`
+		} `json:"error"`
+		Result hexutil.Bytes `json:"result"`
+	}
+	s.NoError(json.Unmarshal([]byte(result), &r))
+	s.Empty(r.Error.Message)
+	s.NotEmpty(r.Result)
+}
+
+// generateSymKey generates and stores a symetric key.
+func (s *DebugAPISuite) generateSymKey() string {
+	w, err := s.Backend.StatusNode().WhisperService()
+	s.Require().NoError(err)
+	symID, err := w.GenerateSymKey()
+	s.Require().NoError(err)
+
+	resp := s.Backend.CallRPC(`{"jsonrpc":"2.0","method":"shh_addSymKey",
+			"params":["` + symID + `"],
+			"id":1}`)
+	type returnedIDResponse struct {
+		Result string
+		Error  interface{}
+	}
+	symkeyAddResp := returnedIDResponse{}
+	err = json.Unmarshal([]byte(resp), &symkeyAddResp)
+	s.NoError(err)
+
+	return symID
+}
+
+// sendPostConfirmMessage calls debug_postSync endpoint with valid
+// parameters.
+func (s *DebugAPISuite) sendPostConfirmMessage(symID string) string {
+	req := whisper.NewMessage{
+		SymKeyID:  symID,
+		PowTarget: whisper.DefaultMinimumPoW,
+		PowTime:   200,
+		Topic:     whisper.TopicType{0x01, 0x01, 0x01, 0x01},
+		Payload:   []byte("hello"),
+	}
+	body, _ := json.Marshal(req)
+
+	basicCall := fmt.Sprintf(
+		`{"jsonrpc":"2.0","method":"debug_postSync","params":[%s],"id":67}`,
+		body)
+
+	return s.Backend.CallPrivateRPC(basicCall)
```

Comment:
If we use `CallPrivateRPC`, can't we mark the API as `Debug: false` in https://github.com/status-im/status-go/blob/f50b439a63e4686c4319e0e03f658da2228d1c2c/services/shhext/service.go#L86?

Reply:
This endpoint is intended to be used by `status-scale` which creates a cluster based on docker. So, we may found ourselves accessing this endpoint through `http`.